engine/src/main/java/io/zeebe/engine/processor/AsyncSnapshotDirector.java

```diff
@@ -156,6 +155,12 @@ private void prepareTakingSnapshot() {
         futureLastProcessedPosition,
         (lastProcessedPosition, error) -> {
           if (error == null) {
+            if (lastProcessedPosition == StreamProcessor.UNSET_POSITION) {
+              LOG.debug("Skip taking snapshot due to no even having ever been processed");
```

Comment:
If we don't skip -1, then we will try to look up the entry at position -1, which doesn't exist, and before this was triggering an NPE (as it was returning a nullable type). Now it would be handled properly anyway, but it would write a weird warning that we couldn't find the position in the log. This seems nicer for the user.

Reply:
Was not aware of the lookup actually, seems to be new ?

Reply:
Since we migrated to use a single snapshot store - Atomix snapshot metadata needs the entry index, so we have to look that up when we take a snapshot.

Reply:
ok thanks